project/src/main/scala/ScapegoatInspectionsGenerator.scala

```diff
@@ -0,0 +1,100 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ */
+import java.nio.file.Paths
+import com.sksamuel.scapegoat.Inspection
+import io.github.classgraph.ClassGraph
+import sbt.Keys._
+import sbt._
+
+import scala.meta._
+import scala.collection.JavaConverters._
+
+/** SBT Task that generates a managed file with all scapegoat inspections */
+object ScapegoatInspectionsGenerator {
+
+  /** Scapegoat inspections that won't be included in the generated file */
+  val BlacklistedInspections = Set(
+    "com.sksamuel.scapegoat.inspections.collections.FilterDotSizeComparison", // Not implemented yet
+    "com.sksamuel.scapegoat.inspections.collections.ListTail" // Not implemented yet
+  )
+
+  val generatorTask = Def.task {
+    val log = streams.value.log
+    log.info("Generating the scapegoat inspections file.")
+
+    // Load the template file.
+    val templateFile = Paths
+      .get(
+        baseDirectory.value.toString,
+        "project",
+        "src",
+        "main",
+        "resources",
+        "ScapegoatInspections.scala"
+      )
+
+    val allScapegoatInspections = extractInspections()
+    val stringifiedScapegoatIsnpections = stringifyInspections(allScapegoatInspections)
+    val transformed = fillTemplate(templateFile.parse[Source].get, stringifiedScapegoatIsnpections)
+
+    val scapegoatInspectionsFile = (sourceManaged in Compile).value / "scapegoat" / "inspections.scala"
+    IO.write(scapegoatInspectionsFile, transformed.syntax)
+
+    Seq(scapegoatInspectionsFile)
+  }
+
+  /** Returns all scapegoat inspections, except the ones that should be ignored */
+  def extractInspections(): List[(String, Inspection)] =
+    // We need to override the scanner class loader,
+    // so it can cast the loaded classes
+    // from class[_] to class[Inspection]
+    new ClassGraph()
+      .overrideClassLoaders(classOf[Inspection].getClassLoader)
+      .whitelistPackages("com.sksamuel.scapegoat.inspections")
+      .scan()
+      .getSubclasses("com.sksamuel.scapegoat.Inspection")
+      .loadClasses(classOf[Inspection])
+      .asScala
+      .toList
+      .collect {
+        case clazz if !BlacklistedInspections.contains(clazz.getName) =>
+          (clazz.getName, clazz.newInstance())
+      }
+
+  /** Stringifies a list of scapegoat inspections */
+  def stringifyInspections(scapegoatInspections: List[(String, Inspection)]): List[String] =
+    scapegoatInspections map {
+      case (inspectionClassName, inspection) =>
+        s"""ScapegoatInspection(
+           |  id = "$inspectionClassName",
+           |  name = "${inspection.text}",
+           |  description = "${inspection.explanation.getOrElse("No Explanation")}",
```

Comment:
I can't remember if I made a comment about this when you first implemented this, but maybe we should just leave the description blank (or just leave it as `None`) if it's missing instead of saying `No Explanation` - what do you think?

Reply:
Uhm, Well I can change the type of that field from `String` to `Option[String]`, and leave the resposability of handling missing descriptions to the **RulesRepository**, that makes sense.
But what do you think would be a good _default_ value?, I agree "No Explanation" was not a good one; it was intendeed as a temporary palce holder but I forgot to change it. I'm not a fan of leaving it empty.

Reply:
> Uhm, Well I can change the type of that field from String to Option[String], and leave the resposability of handling missing descriptions to the RulesRepository, that makes sense.

👍 

> But what do you think would be a good default value?, I agree "No Explanation" was not a good one; it was intendeed as a temporary palce holder but I forgot to change it. I'm not a fan of leaving it empty.

TBH, I'd be tempted to just leave it empty. If scapegoat doesn't provide any description then there isn't much we can do here. I've opened another issue some time ago (#87) which is relevant to this, perhaps we should tackle this separately and for now leave the description empty if it's empty.

Reply:
Roger